nomos-services/storage/src/backends/surrealdb.rs

```diff
@@ -0,0 +1,171 @@
+use crate::backends::{StorageBackend, StorageSerde, StorageTransaction};
+use async_trait::async_trait;
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+use std::marker::PhantomData;
+use std::str::from_utf8;
+use surrealdb::engine::any::{connect, Any};
+use surrealdb::Surreal;
+use tokio::runtime::Handle;
+use tokio::task::block_in_place;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Data {
+    pub data: Bytes,
+}
+
+impl Data {
+    fn new(value: Bytes) -> Self {
+        Self { data: value }
+    }
+
+    fn to_bytes(&self) -> Bytes {
+        self.data.clone()
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Surreal(#[from] surrealdb::Error),
+    #[error(transparent)]
+    Utf8(#[from] std::str::Utf8Error),
+}
+
+#[derive(Clone)]
+pub struct SurrealBackendSettings {
+    pub db_path: String,
+    pub namespace: String,
+    pub database: String,
+}
+
+impl SurrealBackendSettings {
+    pub fn new(db_path: String, namespace: String, database: String) -> Self {
+        Self {
+            db_path,
+            namespace,
+            database,
+        }
+    }
```

Comment:
Nipick. You can actually make it so it can consume other stuff into the string itself.
```suggestion
    pub fn new(db_path: impl Into<String>, namespace: impl Into<String>, database: impl Into<String>) -> Self {
        Self {
            db_path: db_path.into(),
            namespace: namespace.into(),
            database: database.into(),
        }
    }
```

That way later on you do not need to do it manually: `String::from`.

Reply:
Cool, thanks :)